### PR TITLE
Merge Shopper Insights module into Main

### PR DESCRIPTION
**Pull request by tdchow**
### Summary of changes
 - Merge the `payment-insights-complete-feature` feature branch into `main`

Note: this PR should be merged with a merge commit and not a squash to preserve commit history.

### PayPal and Braintree engineering requirements
 > If you're an engineer for PayPal or Braintree, complete this section. For PRs from the community, please ignore!

**When do these changes need to be released?** 
[ ] ASAP (let's discuss outside this PR)
[ ] Soon, here's our target release date: <DD/MM/YYYY>
[ ] Whenever, no rush

### CODE REVIEW

File: PayPal/src/test/java/com/braintreepayments/api/PayPalVaultRequestUnitTest.java [hunk 1]
```diff
@@ -80,17 +84,33 @@ public void parcelsCorrectly() {
...
     }
+
+    @Test
+    public void createRequestBody_sets_userAuthenticationEmail_when_not_null() throws JSONException {
+        String payerEmail = "payer_email@example.com";
+        PayPalVaultRequest request = new PayPalVaultRequest();
+
+        request.setUserAuthenticationEmail(payerEmail);
+        String requestBody = request.createRequestBody(
+            mock(Configuration.class),
```
Comment (sshropshire) [inline PayPal/src/test/java/com/braintreepayments/api/PayPalVaultRequestUnitTest.java:108]: 🧶 : We do have some canned `Authorization` and `Configuration` strings in the `Fixtures` if we ever want to reduce mocks, but these work well here. Just a [callout](https://github.com/braintree/braintree_android/blob/a25c8122e427bc7e6a2f43b68b9cc10a5d6602e2/BraintreeCore/src/test/java/com/braintreepayments/api/ApiClientUnitTest.kt#L29) to a pattern that exists in the codebase.

**Review by sshropshire — APPROVED** [review]

**Review by saperi22 — APPROVED** [review]

**Review by warmkesselj — APPROVED** [review]

File: ShopperInsights/src/main/java/com/braintreepayments/api/ShopperInsightsClient.kt [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+                error = IllegalArgumentException(
+                    "One of ShopperInsightsRequest.email or ShopperInsightsRequest.phone must be " +
+                            "non-null."
+                )
+            )
+            return
+        }
+
+        // TODO: get correct merchant ID from SDK
+        val merchantId = "MXSJ4F5BADVNS"
```
Comment (sarahkoop) [inline ShopperInsights/src/main/java/com/braintreepayments/api/ShopperInsightsClient.kt:58]: Should this be updated based on the TODO?

Comment (tdchow) [reply ShopperInsights/src/main/java/com/braintreepayments/api/ShopperInsightsClient.kt:58]: Good catch, yep it should be updated. Do you have a preference on if we update it as a part of this PR or a follow-up PR into main?

File: PayPal/src/test/java/com/braintreepayments/api/PayPalVaultRequestUnitTest.java (hunk 1 above)
Comment (tdchow) [reply PayPal/src/test/java/com/braintreepayments/api/PayPalVaultRequestUnitTest.java:108]: ohh nice! i don't think we were aware of the `Fixtures` when writing these tests. we can align this once the feature is out the door 👍

File: ShopperInsights/src/main/java/com/braintreepayments/api/ShopperInsightsClient.kt (hunk 2 above)
Comment (tdchow) [reply ShopperInsights/src/main/java/com/braintreepayments/api/ShopperInsightsClient.kt:58]: Actually, i realized this should be removed. We're no longer sending the merchant ID as a param in the request body. The backend determines the merchant based on the auth fingerprint.

File: PayPal/src/test/java/com/braintreepayments/api/PayPalVaultRequestUnitTest.java (hunk 1 above)
Comment (sshropshire) [reply PayPal/src/test/java/com/braintreepayments/api/PayPalVaultRequestUnitTest.java:108]: Yeah definitely this is good now.

File: ShopperInsights/src/main/java/com/braintreepayments/api/ShopperInsightsClient.kt (hunk 2 above)
Comment (jaxdesmarais) [reply ShopperInsights/src/main/java/com/braintreepayments/api/ShopperInsightsClient.kt:58]: > Good catch, yep it should be updated. Do you have a preference on if we update it as a part of this PR or a follow-up PR into main?

I think any changes should go in a new PR vs making the change directly in the feature branch PR before merging this branch into `main`. Generally our stance is that we do not require approvals on merging feature branches into `main` as all PRs that make up the feature branch should have been reviewed and received 2 approvals already. Opening the feature branch PR to `main` is mainly just a check that CI is all still green more than anything else.

**Comment by scannillo** [comment]
Closing due to pushed back timeline for releasing ShopperInsights feature on BT Mobile SDKs.

**Pull request closed without merging**